test/integration/builder/additional.js

```diff
@@ -256,36 +256,50 @@ module.exports = function(knex) {
     });
 
 
-    it('.timeout() should throw TimeoutError', function() {
+    it('.timeout() should throw TimeoutError', function(done) {
       var dialect = knex.client.config.dialect;
-      var rows = _.times(dialect === 'sqlite3' ? 100 : 3000, function() {
-        return {
-          id: 1,
-          value: 'rO8F8YrFS6uoivuRiVnwrO8F8YrFS6uoivuRiVnwuoivuRiVnw'
-        };
-      });
+      if(dialect === 'sqlite3') { done(); } //TODO -- No built-in support for sleeps
+      var testQueries = {
+        'postgres': function() {
+          return knex.raw('SELECT pg_sleep(5)');
+        },
+        'mysql': function() {
+          return knex.raw('SELECT SLEEP(5)');
+        },
+        'mysql2': function() {
+          return knex.raw('SELECT SLEEP(5)');
+        },
+        maria: function() {
+          return knex.raw('SELECT SLEEP(5)');
+        },
+        mssql: function() {
+          return knex.raw('WAITFOR DELAY \'00:00:05\'');
+        },
+        oracle: function() {
+          return knex.raw('dbms_lock.sleep(5)');
+        },
+        'strong-oracle': function() {
+          return knex.raw('dbms_lock.sleep(5)');
+        }
+      };
```

Comment:
Once you have this completely working, assuming this is the way this gets implemented, might be worth extracting this into an engine specific test utility layer so there is one place where such engine specific information needs to be tweaked for each engine instead of hunting all such cases through all the tests.

One other such thing that pops to mind is a way to say whether a specific engine reports its BEGIN & ROLLBACK statements as queries, i.e. using the `query` event, which `oracle` & `mssql` do not seem to do.